lib/query_components/core_query.rb

```diff
@@ -47,7 +56,9 @@ def payload
       }
     end
 
-    private
+    def payload_for_quoted_phrase
+      exact_field_boosts
```

Comment:
I think this should be changed to produce exactly the same query as the version in `text_query` does in this case.  The query you're producing here uses fields which have had stop wording applied, so won't work quite right.